### ISSUE TITLE
Disconnected from k8s-container

### ISSUE THREAD

**Issue by tmartin-gh** [microsoft/vscode-remote-release#2814]
TL;DR: Using Explorer -> Open Folder fails on VSCode attached to pod with error "Disconnected from k8s-container+podname=<my pod name>+<some other pod parameters".

I am running VSCode with the Remote-Container and Kubernetes plugins on a Linux machine that has access to a custom-rolled (mostly kubespray) on-prem/bare-metal Kubernetes cluster.  Pods deployed on the cluster do NOT have internet access.

When I use the Kubernetes plugin on the local VSCode winidow to "Attach Visual Studio Code" to my pod, a second VSCode window opens and shows "Opening Remote" and then after a few seconds the pod name appears.  When I then use Explorer -> Open Folder to pick a folder on the remote pod's filesystem (which is just an NFS mount of a folder on my local file system) I see another "Opening Remote" followed by "Disconnected from k8s-container+podname=<my pod name>+<some other pod parameters".

This is my first VSCode bug report, not sure how to enable the debug console to see if there is other more useful info.  If someone can provide guidance I will pull more debug info...

I put this in my local settings.json and attempted to do it also in the remote settings.json:
```
{
    "remote.containers.logLevel": "trace"
} 
```

But I still don't see anything show up in the "Output" or "Debug Console" views. (Dumb question - Is there another spot to look?)

I'm guessing it might just have something to do with the pod having lack of internet access, or something not quite right in the container setup.

- VSCode Version:
Version: 1.44.2
Commit: ff915844119ce9485abfe8aa9076ec76b5300ddd
Date: 2020-04-16T17:50:03.709Z

- Local OS Version:
Electron: 7.1.11
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Linux x64 4.15.0-43-generic
(Ubuntu 18.04)

- Remote OS Version (OS running on k8s host node where the pod is deployed)
Should be Ubuntu 18.04 

- Pod OS
FROM ubuntu:18.04 (well, a derivative, several apt installs)

Pod spec:
`
    spec:
      securityContext:
         runAsUser: <my uid on the local Linux server>
         runAsGroup: <my gid on the local Linux server>
      containers:
      - image: <my image URL>
        imagePullPolicy: Always
        name: <my container name>
        tty: true
        volumeMounts:   
        - mountPath: /home/tmartin
          name: nfs-mount
        resources:
          requests:
            memory: "8Gi"
            cpu: "4"
            ephemeral-storage: "1Gi"
          limits:
            memory: "32Gi"
            cpu: "16"     
            ephemeral-storage: "2Gi"     
      imagePullSecrets:
      - name: <my secret name>
      priorityClassName: <my priority class>
      restartPolicy: Always
      volumes:
      - name: nfs-mount
        persistentVolumeClaim:
          claimName: <my volume claim name>
`

**Comment by tmartin-gh** [comment]
Update - I was able to find a slightly more verbose error message:
`Failed to connect to the remote extension host server (Error: Cannot destructure property 'contents' of 'this._folderConfigurations.get(...) as it is undefined.`

I'm positive this is because the Pod isn't able to connect directly to the internet.  Is there a way to pre-install VSCode extensions on the container image when I build it?  `apt install vscode-remote-containers` or something of the sort?

**Comment by jeremysf** [comment]
I am experiencing the same issue, although in my case, I'm trying to attach to a container in a pod running in a local Kubernetes cluster (kind based).

I tried using the container from:

https://github.com/Microsoft/vscode-remote-try-go

and I can see the `~/.vscode-server` and am able to pull my repository, so the git credential sharing is working, etc.

But I have the same issue as @tmartin-gh where if I try to open a folder on the container, I am able to browse to it but then I get:

```Failed to connect to the remote extension host server (Error: Cannot destructure property 'contents' of 'this._folderConfigurations.get(...) as it is undefined.```

**Comment by artificial-aidan** [comment]
Seeing the same issue still. But found a weird workaround that might lead to some answers. This is the workaround.

1. Attach Visual Studio to a container in a pod (succeeds)
2. Open a folder in a container (fails)
3. Now with the disconnected window, you should see the first item in the `Recent` list on the welcome page. And it should be the folder you just tried to open. Open that link
4. Now you are in a weird state that I don't totally understand. But from here, try to open the folder you opened before again.
5. It should work now.
